src/box/lua/config/applier/credentials.lua

```diff
@@ -526,6 +513,37 @@ local function apply(config)
     create_users(credentials.users)
 end
 
+local function wait_rw(config)
+    if box.info.ro then
+        log.verbose('credentials.apply: Tarantool is in Read Only mode ' ..
+                    'credentials will be set up in the background when it ' ..
+                    'is switched to Read Write mode')
+        box.ctl.wait_rw()
+    end
+    set_credentials(config)
+end
+
+local function apply(config)
+    -- Fiber is required here to have the credentials synced in the background
+    -- when Tarantool is in Read Only mode after it is switched to RW.
+
+    -- Note that only one fiber exists at a time.
+    if fiber_wait_rw == nil or fiber_wait_rw:status() == 'dead' then
+        fiber_wait_rw = fiber.new(wait_rw, config)
+        fiber_wait_rw:set_joinable(true)
+    end
+
+    -- If Tarantool is already in Read Write mode, credentials are still
+    -- applied in the fiber to avoid possible concurrency issues, but the
+    -- main thread is blocked by the `join()`.
```

Comment:
Now, with `set_joinable()` inside if clause, the comment is legit. We block the main thread when Tarantool is already in RW.

Reply:
The fiber that processes the configuration applying is blocked, that's right. I guess that it is a kind of the main fiber in context of startup. However, if config:reload() is invoked, it is the fiber that processes the call. If an update comes from etcd, this code will be executed in a fiber that watches for etcd updates.

But! I want to highlight that TX **thread** is not blocked. Other fibers can work within the thread, while this fiber waits for `:join()`.

Reply:
Oh, right. Reworded:
```
      -- If Tarantool is already in Read Write mode, credentials are still
      -- applied in the fiber to avoid possible concurrency issues, but the
      -- main applier fiber is blocked by the `join()`.
```